src/Shared/Reporting/ColorFormatter.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Tools.Internal
+{
+    public class ColorFormatter : IFormatter
```

Comment:
Yeah. It has an `IConsole` API internally. The API was insufficient because it didn't provide access to `Console.In`. 

Follow up: we remove duplication by putting this API and others in a common location to be used by Logging.Console and here as well? Could help with https://github.com/aspnet/Logging/issues/491